src/commands/Minion/craft.ts

```diff
@@ -78,75 +76,77 @@ export default class extends BotCommand {
-					if (userGP < qty) {
-						return msg.send('You do not have enough GP.');
-					}
-					quantity = Math.min(quantity, Math.floor(userGP / qty));
-					continue;
-				}
-				const itemsOwned = userBank[parseInt(itemID)];
-				if (itemsOwned < qty) {
-					return msg.send(`You dont have enough ${itemNameFromID(parseInt(itemID))}.`);
-				}
-				quantity = Math.min(quantity, Math.floor(itemsOwned / qty));
-			}
+			const max = userBank.fits(craftable.inputItems);
+			if (max < quantity && max !== 0) quantity = max;
 		}
 
 		const duration = quantity * timeToCraftSingleItem;
-
 		if (duration > maxTripLength) {
 			return msg.send(
 				`${msg.author.minionName} can't go on trips longer than ${formatDuration(
 					maxTripLength
-				)}, try a lower quantity. The highest amount of ${Craft.name}s you can craft is ${Math.floor(
+				)}, try a lower quantity. The highest amount of ${craftable.name}s you can craft is ${Math.floor(
 					maxTripLength / timeToCraftSingleItem
 				)}.`
 			);
 		}
 
-		// Check the user has add the required items to craft.
-		for (const [itemID, qty] of requiredItems) {
-			const id = parseInt(itemID);
-			if (id === 995) {
-				const userGP = msg.author.settings.get(UserSettings.GP);
-				if (userGP < qty * quantity) {
-					return msg.send(`You don't have enough ${itemNameFromID(id)}.`);
-				}
-				continue;
-			}
-			if (!bankHasItem(userBank, id, qty * quantity)) {
-				return msg.send(`You don't have enough ${itemNameFromID(id)}.`);
-			}
+		const itemsNeeded = craftable.inputItems.clone().multiply(quantity);
+		let gpNeeded = 0;
+		const currentGP = msg.author.settings.get(UserSettings.GP);
+		if (itemsNeeded.has('Coins')) {
+			gpNeeded = itemsNeeded.amount('Coins');
+			itemsNeeded.remove('Coins');
+		}
+
+		// Check the user has all the required items to craft.
+		if (!userBank.has(itemsNeeded.bank)) {
+			return msg.send(
+				`You don't have enough items. For ${quantity}x ${craftable.name}, you're missing **${itemsNeeded
+					.clone()
+					.remove(userBank)}**.`
+			);
 		}
 
-		// Remove the required items from their bank.
-		let newBank = { ...userBank };
-		for (const [itemID, qty] of requiredItems) {
-			if (parseInt(itemID) === 995) {
-				await msg.author.removeGP(qty * quantity);
-				continue;
-			}
-			newBank = removeItemFromBank(newBank, parseInt(itemID), qty * quantity);
+		// Check the user has the GP to craft.
+		if (gpNeeded > currentGP) {
+			return msg.send(
+				`You don't have enough GP. For ${quantity}x ${craftable.name}, you're missing **${
+					gpNeeded - currentGP
+				}**.`
+			);
 		}
-		await msg.author.settings.update(UserSettings.Bank, newBank);
+
+		await msg.author.removeItemsFromBank(itemsNeeded);
+
+		if (gpNeeded > 0) await msg.author.removeGP(gpNeeded);
+
+		await this.client.settings.update(
+			ClientSettings.EconomyStats.CraftingCostBank,
+			addBanks([this.client.settings.get(ClientSettings.EconomyStats.CraftingCostBank), itemsNeeded.bank])
+		);
```

Comment:
This should use `updateBankSetting`, you were probably looking at a file which is still doing it in this way instead of with the function i made for this.